contracts/SwapAndBridge.sol

```diff
@@ -291,4 +328,174 @@ contract UniversalSwapAndBridge is SwapAndBridgeBase {
             acrossInputToken
         );
     }
+
+    /**
+     * @notice Swaps an EIP-2612 token on this chain via specified router before submitting Across deposit atomically.
+     * Caller can specify their slippage tolerance for the swap and Across deposit params.
+     * @dev If swapToken does not implement `permit` to the specifications of EIP-2612, this function will fail.
+     * @param swapToken Address of the token that will be swapped for acrossInputToken.
+     * @param acrossInputToken Address of the token that will be bridged via Across as the inputToken.
+     * @param routerCalldata ABI encoded function data to call on router. Should form a swap of swapToken for
+     * enough of acrossInputToken, otherwise this function will revert.
+     * @param swapTokenAmount Amount of swapToken to swap for a minimum amount of depositData.inputToken.
+     * @param minExpectedInputTokenAmount Minimum amount of received depositData.inputToken that we'll submit bridge
+     * deposit with.
+     * @param depositData Specifies the Across deposit params we'll send after the swap.
+     * @param deadline Deadline before which the permit signature is valid.
+     * @param v v of the permit signature.
+     * @param r r of the permit signature.
+     * @param s s of the permit signature.
+     */
+    function swapAndBridgeWithPermit(
+        IERC20Permit swapToken,
+        IERC20 acrossInputToken,
+        bytes calldata routerCalldata,
+        uint256 swapTokenAmount,
+        uint256 minExpectedInputTokenAmount,
+        DepositData calldata depositData,
+        uint256 deadline,
+        uint8 v,
+        bytes32 r,
+        bytes32 s
+    ) external nonReentrant {
+        IERC20 _swapToken = IERC20(address(swapToken)); // Cast IERC20Permit to IERC20.
+        if (_swapToken.allowance(msg.sender, address(this)) < swapTokenAmount)
```

Comment:
Why should we be checking for front-running with permit but not with transferFrom? I'd argue we'd want the same allowance() check in both swapAndBridge AND swapAndBridgeWithPermit.

But I favor having no checks at all and just letting the subsequent safeTransferFrom revert

Reply:
I thought the front-running was with respect to somebody calling `permit` on the token before this transaction is executed, so the approval would be there, but the call to permit would revert since the signature input in this function would have already been used.

Reply:
is this something that would happen in the normal course of transactions?

Reply:
It's only possible if people want to grief users. Updated here: https://github.com/across-protocol/contracts/pull/751/commits/ab96160a7580ed6a0bfb2aabe08b9b14ef16ded5